src/t_string.c

```diff
@@ -593,15 +597,17 @@ void incrDecrCommand(client *c, long long incr) {
     } else {
         new = createStringObjectFromLongLongForValue(value);
         if (o) {
-            dbReplaceValue(c->db,c->argv[1],new);
+            dbReplaceValueWithDictEntry(c->db,c->argv[1],new,de);
         } else {
             dbAdd(c->db,c->argv[1],new);
         }
     }
     signalModifiedKey(c,c->db,c->argv[1]);
     notifyKeyspaceEvent(NOTIFY_STRING,"incrby",c->argv[1],c->db->id);
     server.dirty++;
-    addReplyLongLong(c, value);
+    addReplyProto(c,":",1);
+    addReply(c,new);
+    addReplyProto(c,"\r\n",2);
```

Comment:
it's a bit ugly to compose the protocol here directly. do we do that on other commands?
maybe instead add a new `addReplyLongLongFromStr`?

Reply:
resolved in 13eb4f4add0a0309b2af0945e52d539294ead9a2